_appmap/utils.py

```diff
@@ -222,3 +222,27 @@ def scenario_filename(name, separator="_"):
     pattern = r"[^a-z0-9\-_]+"
     replacement = separator
     return re.sub(pattern, replacement, name, flags=re.IGNORECASE)
+
+
+def locate_file_up(filename, start_dir=None, stop_dir=None):
+    """
+    Search for a file in the current directory and recursively up to the root directory.
+    
+    :param filename: The name of the file to locate.
+    :param start_dir: The directory to start the search from. Defaults to the current.
+    :param stop_idr: The directory to stıp the search. If None search is performed until
+                     the root of the file system.
+    :return: The path to the directory containing the file or None if the file cannot be found.
+    """
+    if start_dir is None:
+        start_dir = os.getcwd()
+
+    file_path = os.path.join(start_dir, filename)
```

Comment:
In general, it seems like it might be better to use `pathlib`. Was there a reason you chose `os.path` here?